src/addon-manager-proxy.ts

```diff
@@ -11,32 +11,51 @@
 
 'use strict';
 
-const {APIRequest, APIResponse} = require('./api-handler');
-const {MessageType} = require('./constants');
+import { Action } from "./action";
+import { Adapter } from "./adapter";
+import { APIHandler } from "./api-handler";
+import { Device } from "./device";
+import { Event } from "./event";
+import { Notifier } from "./notifier";
+import { Outlet } from "./outlet";
+import { PluginClient } from "./plugin-client";
+import { Property } from "./property";
+
+const { APIRequest, APIResponse } = require('./api-handler');
+const { MessageType } = require('./constants');
 const EventEmitter = require('events').EventEmitter;
 
-class AddonManagerProxy extends EventEmitter {
-  constructor(pluginClient, {verbose} = {}) {
+interface MockAdapter {
+  clearState(): Promise<void>;
+  addDevice(deviceId: string, deviceDescr: any): Promise<any>;
```

Comment:
@mrstegeman Could you please help us with figuring out what is wrapped by promise? Here and one line below.

Reply:
This is something that's used for integration testing only.